native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/NativeCompileNoForkMojo.java

```diff
@@ -101,9 +115,77 @@ public void execute() throws MojoExecutionException {
         maybeSetMainClassFromPlugin(this::consumeConfigurationNodeValue, "org.apache.maven.plugins:maven-assembly-plugin", "archive", "manifest", "mainClass");
         maybeSetMainClassFromPlugin(this::consumeConfigurationNodeValue, "org.apache.maven.plugins:maven-jar-plugin", "archive", "manifest", "mainClass");
         maybeAddGeneratedResourcesConfig(buildArgs);
+
+        generateAugmentedSBOMIfNeeded();
+
         buildImage();
     }
 
+    /**
+     * Generates an augmented SBOM using the {@link SBOMGenerator} based on specific conditions:
+     *
+     * 1. If {@link NativeCompileNoForkMojo#augmentedSBOM} is explicitly set to false: No SBOM is generated.
+     * 2. If {@link NativeCompileNoForkMojo#augmentedSBOM} is explicitly set to true: An augmented SBOM is
+     *    generated if the required conditions are met.
+     * 3. If {@link NativeCompileNoForkMojo#augmentedSBOM} is not set: An augmented SBOM is generated only if
+     *    SBOM generation is configured for Native Image via a build argument.
+     *
+     * Note: Augmented SBOMs are only supported in Oracle GraalVM for JDK {@link SBOMGenerator#requiredNativeImageVersion}
+     * or later.
+     *
+     * @throws IllegalArgumentException if {@link NativeCompileNoForkMojo#augmentedSBOM} is explicitly set to true
+     *         but required conditions are not met (e.g., using community edition or JDK version 23 or earlier).
+     * @throws MojoExecutionException if augmented SBOM generation was attempted but failed.
+     */
+    private void generateAugmentedSBOMIfNeeded() throws IllegalArgumentException, MojoExecutionException {
+        boolean optionWasSet = augmentedSBOM != null;
+        augmentedSBOM = optionWasSet ? augmentedSBOM : true;
+
+        int detectedJDKVersion = NativeImageUtils.getMajorJDKVersion(getVersionInformation());
+        String sbomNativeImageFlag = "--enable-sbom";
+        boolean sbomEnabledForNativeImage = getBuildArgs().stream().anyMatch(v -> v.contains(sbomNativeImageFlag));
+        if (optionWasSet) {
```

Comment:
:question: Can you clarify the intent here? Why would we have to set both `--enable-sbom` explicitly and augmented sbom flag?

Reply:
Setting both `--enable-sbom` and the augmented sbom flag is not required. We check `optionWasSet` since the behavior is dependent on if the user explicitly set the augmented flag option. I tried to make the intent clear via the javadoc, variable names, and comments. What exactly was unclear?